packages/shared/src/components/GrowthBookProvider.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {
+  ReactElement,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useMutation } from 'react-query';
+import {
+  Context,
+  GrowthBook,
+  GrowthBookProvider as Provider,
+} from '../lib/growthbook';
+import { isProduction } from '../lib/constants';
+import { BootApp, BootCacheData } from '../lib/boot';
+import { decrypt } from './crypto';
+import { apiUrl } from '../lib/config';
+
+export type GrowthBookProviderProps = {
+  app: BootApp;
+  user: BootCacheData['user'];
+  deviceId: string;
+  version?: string;
+  experimentation?: {
+    f: string;
+    e: string[];
+  };
+  fetchMethod?: typeof fetch;
+  children?: ReactNode;
+};
+
+export const GrowthBookProvider = ({
+  app,
+  user,
+  deviceId,
+  version,
+  experimentation,
+  fetchMethod = fetch,
+  children,
+}: GrowthBookProviderProps): ReactElement => {
+  const { mutateAsync: sendAllocation } = useMutation(
```

Comment:
Allocation is basically adding user to experiment?

Reply:
I'm also curious about this one, what does it entail?

Reply:
Yes exactly